packages/block-library/src/comments-title/edit.js

```diff
@@ -136,78 +106,52 @@ export default function Edit( {
 
 	const postTitle = isSiteEditor ? __( '"Post Title"' ) : `"${ rawTitle }"`;
 
-	const singlePlaceholder = showPostTitle
-		? __( 'One response to ' )
-		: __( 'One response' );
-
-	const singlePlaceholderNoCount = showPostTitle
-		? __( 'Response to ' )
-		: __( 'Response' );
-
-	const multiplePlaceholder = showPostTitle
-		? __( 'responses to ' )
-		: __( 'responses' );
-
-	const multiplePlaceholderNoCount = showPostTitle
-		? __( 'Responses to ' )
-		: __( 'Responses' );
+	let placeholder;
+	if ( showCommentsCount && commentsCount !== undefined ) {
+		if ( showPostTitle ) {
+			if ( commentsCount === 1 ) {
+				/* translators: %s: Post title. */
+				placeholder = sprintf( __( 'One response to %s' ), postTitle );
+			} else {
+				placeholder = sprintf(
+					/* translators: 1: Number of comments, 2: Post title. */
+					_n(
+						'%1$s response to %2$s',
+						'%1$s responses to %2$s',
+						commentsCount
+					),
+					commentsCount,
+					postTitle
+				);
+			}
+		} else if ( commentsCount === 1 ) {
+			placeholder = __( 'One response' );
+		} else {
+			placeholder = sprintf(
+				/* translators: %s: Number of comments. */
+				_n( '%s responses', '%s responses', commentsCount ),
```

Comment:
String doesn't exist in Core yet. However, this is arguably needed to fix the issue at the core of this PR (use `sprintf` with placeholders).